src/main/java/Duke.java

```diff
@@ -8,7 +8,9 @@ public static void main(String[] args) {
         System.out.println("____________________________________________________________");
 
         Scanner scanner = new Scanner(System.in);
-        ArrayList<Task> tasks = new ArrayList<>();
+        Task[] tasks = new Task[100];
```

Comment:
Instead of limiting the max size of task to be 100. Maybe what you can do is to hash the array to a bigger array whenever the task no. goes beyond 100. i.e. add the rehash function.